apps/meteor/tests/unit/server/modules/watchers/lib/messages.spec.ts

```diff
@@ -0,0 +1,224 @@
+		let originalEnv: NodeJS.ProcessEnv;
+
+		beforeEach(() => {
+			originalEnv = { ...process.env };
+			sinon.resetHistory();
+		});
+
+		afterEach(() => {
+			process.env = originalEnv;
+		});
+
+		const testCases = [
+			{
+				description: 'should return undefined if message is hidden or imported',
+				message: { ...sampleMessage, _hidden: true },
+				hideSystemMessages: [],
+				useRealName: false,
+				expectedResult: undefined,
+			},
+			{
+				description: 'should hide message if type is in hideSystemMessage settings',
+				message: sampleMessage,
+				hideSystemMessages: ['user-muted', 'mute_unmute'],
+				useRealName: false,
+				expectedResult: undefined,
+			},
+			{
+				description: 'should return the message with real name if useRealName is true',
+				message: sampleMessage,
+				hideSystemMessages: [],
+				useRealName: true,
+				expectedResult: { ...sampleMessage, u: { ...sampleMessage.u, name: 'Real User' } },
+			},
+			{
+				description: 'should return the message if Hide_System_Messages is undefined',
+				message: sampleMessage,
+				hideSystemMessages: undefined,
+				useRealName: false,
+				expectedResult: sampleMessage,
+			},
+			{
+				description: 'should return undefined if the message type is muted and a mute_unmute is received',
+				message: { ...sampleMessage, t: 'mute_unmute' },
+				hideSystemMessages: ['user-muted', 'mute_unmute'],
+				useRealName: false,
+				expectedResult: undefined,
+			},
+			{
+				description: 'should return the message if no system messages are muted',
+				message: sampleMessage,
+				hideSystemMessages: [],
+				useRealName: false,
+				expectedResult: sampleMessage,
+			},
+			{
+				description: 'should hide message if type is room-archived',
+				message: { ...sampleMessage, t: 'room-archived' },
+				hideSystemMessages: ['room-archived'],
+				useRealName: false,
+				expectedResult: undefined,
+			},
+			{
+				description: 'should hide message if type is user-unmuted',
+				message: { ...sampleMessage, t: 'user-unmuted' },
+				hideSystemMessages: ['user-unmuted'],
+				useRealName: false,
+				expectedResult: undefined,
+			},
+			{
+				description: 'should hide message if type is subscription-role-added',
+				message: { ...sampleMessage, t: 'subscription-role-added' },
+				hideSystemMessages: ['subscription-role-added'],
+				useRealName: false,
+				expectedResult: undefined,
+			},
+			{
+				description: 'should hide message if type is message_pinned',
+				message: { ...sampleMessage, t: 'message_pinned' },
+				hideSystemMessages: ['message_pinned'],
+				useRealName: false,
+				expectedResult: undefined,
+			},
+			{
+				description: 'should hide message if type is new-owner',
+				message: { ...sampleMessage, t: 'new-owner' },
+				hideSystemMessages: ['new-owner'],
+				useRealName: false,
+				expectedResult: undefined,
+			},
+		];
+
+		testCases.forEach(({ description, message, hideSystemMessages, useRealName, expectedResult }) => {
```

Comment:
iirc there's a test.each or it.each that does this exact thing.

If it's not there on mocha/chai, pls ignore this one